apps/avatax/src/pages/api/webhooks/checkout-calculate-taxes.ts

```diff
@@ -41,6 +51,34 @@ export const checkoutCalculateTaxesSyncWebhook = new SaleorSyncWebhook<Calculate
   webhookPath: "/api/webhooks/checkout-calculate-taxes",
 });
 
+/**
+ * Translate internal errors to responses back to Saleor
+ */
+const webhookErrorToResponseMapper = (req: NextApiRequest, res: NextApiResponse) => {
+  return new Map([
+    [
+      ActiveConnectionServiceErrors.BrokenConfigurationError,
+      () =>
+        res
+          .status(400)
+          .send("App is not configured properly. Please verify configuration or reinstall the app"),
+    ],
+    [
+      ActiveConnectionServiceErrors.MissingMetadataError,
+      () => res.status(400).send("App is not configured properly. Configure the app first"),
+    ],
+    [ActiveConnectionServiceErrors.MissingChannelSlugError, () => res.status(400).send("??? TODO")],
+    [ActiveConnectionServiceErrors.WrongChannelError, () => res.status(500).send("???")],
+    [
+      ActiveConnectionServiceErrors.ProviderNotAssignedToChannelError,
+      () =>
+        res
+          .status(400)
+          .send("App is not configured properly. Please verify configuration or reinstall the app"),
+    ],
+  ]);
```

Comment:
Do you mean those?

```
// apps/avatax/src/modules/taxes/get-active-connection-service.ts
  /**
   * TODO: What does it mean?  How it should behave?
   */
  MissingChannelSlugError: ActiveConnectionServiceError.subclass("MissingChannelSlugError"),

  /**
   * TODO: What does it mean?  How it should behave?
   */
  MissingMetadataError: ActiveConnectionServiceError.subclass("MissingMetadataError"),

  /**
   * TODO: What does it mean?  How it should behave?
   * Should it be handled as BrokenConfigurationError?
   */
  ProviderNotAssignedToChannelError: ActiveConnectionServiceError.subclass(
    "ProviderNotAssignedToChannelError",
  ),
```

If so, then:

- `MissingChannelSlugError` - As far as I can see, it exists because according to some webhooks payloads (e.g. order confirmed, see line 48 in `webhooks/order-confirmed.ts`), the `channelSlug` can be undefined. Not the user's fault.
- `MissingMetadataError`- thrown when the app hasn't yet been configured. It is the user's fault.
- `ProviderNotAssignedToChannelError` - configuration was found, provider was created, but it was not assigned to a channel. User's fault